src/manager/agent.manager.service.ts

```diff
@@ -66,33 +62,56 @@ export class AgentManagerService {
-
-        const containerId = await this.manager.startAgent(agentConfig);
+        try {
+            // Note: according to the docs a 0 value means caching for infinity, however this does not work in practice - it doesn't cache at all
+            // Short term you can pass in -1 and it will cache for max int (ie a very long time)
+            // TODO Long term we should have a proper DB store for permanent agents
+            // (or figure out a way to avoid having to save agent data all together)
+
+            const agentConfig = new AgentConfig(walletId, walletKey, adminApiKey, agentId, agentEndpoint, webhookUrl, adminApiPort, httpPort, seed);
+            const containerId = await this.manager.startAgent(agentConfig);
+
+            // @tothink move this caching to db
+            // adding one second to cache record timeout so that spinDownAgent has time to process before cache deletes the record
+            Logger.info(`record cache limit set to: ${(ttl === 0 ? ttl : ttl + 1)}`);
+            // TODO  we dont really need containerId anymore, could remove it from cache
+            await this.cache.set(agentId, {containerId, adminApiKey, ttl}, {ttl: (ttl === 0 ? ttl : ttl + 1)});
+
+            // ttl = time to live is expected to be in seconds (which we convert to milliseconds).  if 0, then live in eternity
+            if (ttl > 0) {
+                setTimeout(
+                    async () => {
+                        await this.spinDownAgent(agentId);
+                    }, ttl * 1000);
+            }
 
-        // @tothink move this caching to db
-        // adding one second to cache record timeout so that spinDownAgent has time to process before cache deletes the record
-        Logger.info(`record cache limit set to: ${(ttl === 0 ? ttl : ttl + 1)}`);
-        await this.cache.set(agentId, { containerId, adminApiKey, ttl }, {ttl: (ttl === 0 ? ttl : ttl + 1)});
+            let connectionData = {};
+            // when autoConnect is true, then call the createConnection method
+            // when autoCorrect is not defined or null, treat it as true
+            if (autoConnect === true) {
+                // TODO for right now let's delay and then initiate the connection
+                await this.pingConnectionWithRetry(agentId, adminApiPort, adminApiKey, parseInt(process.env.AGENT_RETRY_DURATION, 10));
+                connectionData = await this.createConnection(agentId, adminApiPort, adminApiKey);
+            }
 
-        // ttl = time to live is expected to be in seconds (which we convert to milliseconds).  if 0, then live in eternity
-        if (ttl > 0) {
-            setTimeout(
-                async () => {
-                    await this.spinDownAgent(agentId);
-                }, ttl * 1000);
-        }
+            return {agentId, connectionData};
+        } catch (e) {
+            if (e.statusCode === 409 && e.message.includes('Conflict')) {
```

Comment:
This if statement is specific to docker.  I don't know what we will get in k8s.  I could use your feedback on what to do.  I see a few options:

1. For now, leave it and see what happens in k8s and update when we know more
2. Remove the if statement and on any exception attempt the logic in catch
3. Research how k8s will behave

I'm leaning towards 2.  1 and 3 are a wash for me.

Thoughts?

Reply:
Ya my vote would be #2 as well, since there could potentially be a few different error codes. And it doesn't hurt to try to connect to an existing one, it will fail if it's not there.